cmd/crossplane/main.go

```diff
@@ -17,190 +17,133 @@ limitations under the License.
+		kingpin.FatalIfError(workload.Setup(mgr, log), "Cannot setup workload controllers")
+		kingpin.FatalIfError(mgr.Start(ctrl.SetupSignalHandler()), "Cannot start controller manager")
 
-		// Start the Cmd
-		log.Info("Starting the manager")
-		kingpin.FatalIfError(mgr.Start(signals.SetupSignalHandler()), "Cannot start controller")
 	case extManageCmd.FullCommand():
-		// the "stacks manage" command is being run, the only controllers we should add to the
-		// manager are the stacks controllers
+		log := logging.NewLogrLogger(zl.WithName("stack-manager"))
+		log.Debug("Starting", "sync-period", syncPeriod.String())
+
+		cfg, err := getRestConfig(*extManageTenantKubeconfig)
+		kingpin.FatalIfError(err, "Cannot get config")
 
-		mgr := setupManager(log, syncPeriod, *extManageTenantKubeconfig)
+		mgr, err := ctrl.NewManager(cfg, ctrl.Options{SyncPeriod: syncPeriod})
+		kingpin.FatalIfError(err, "Cannot create manager")
 
-		log.Info("Adding controllers")
-		kingpin.FatalIfError(stacksControllerSetupWithManager(mgr, *extManageHostControllerNamespace), "Cannot add controllers to manager")
+		kingpin.FatalIfError(apis.AddToScheme(mgr.GetScheme()), "Cannot add core Crossplane APIs to scheme")
+		kingpin.FatalIfError(apiextensionsv1beta1.AddToScheme(mgr.GetScheme()), "Cannot add API extensions to scheme")
+		kingpin.FatalIfError(stacks.Setup(mgr, log, *extManageHostControllerNamespace), "Cannot add stacks controllers to manager")
 
-		if *supportTemplates {
-			log.Info("Adding template controllers")
-			kingpin.FatalIfError(stacksTemplateControllerSetupWithManager(mgr), "Cannot add template controllers to manager")
+		if *extManageSupportTemplates {
+			kingpin.FatalIfError(templates.SetupStackConfigurations(mgr, log), "Cannot add stack configuration controller to manager")
+			kingpin.FatalIfError(templates.SetupStackDefinitions(mgr, log), "Cannot add stack definition controller to manager")
 		}
 
-		// Start the Cmd
-		log.Info("Starting the manager")
-		kingpin.FatalIfError(mgr.Start(signals.SetupSignalHandler()), "Cannot start controller")
+		kingpin.FatalIfError(mgr.Start(ctrl.SetupSignalHandler()), "Cannot start controller manager")
+
 	case extUnpackCmd.FullCommand():
-		var outFile io.StringWriter
-		// stack unpack command was called, run the stack unpacking logic
-		if extUnpackOutfile == nil || *extUnpackOutfile == "" {
-			outFile = os.Stdout
-		} else {
-			openFile, err := os.Create(*extUnpackOutfile)
-			kingpin.FatalIfError(err, "Cannot create outfile")
-			defer closeOrError(openFile)
-			outFile = openFile
+		log := logging.NewLogrLogger(zl.WithName("stacks"))
+
+		outFile := os.Stdout
+		if *extUnpackOutfile != "" {
+			f, err := os.Create(*extUnpackOutfile)
+			kingpin.FatalIfError(err, "Cannot create output file")
+			defer kingpin.FatalIfError(f.Close(), "Cannot close file")
+			outFile = f
 		}
+		log.Info("Unpacking stack", "to", outFile.Name())
```

Comment:
Justed tested in host aware mode and stack installation failed due to this line with:

```
failed to parse output from job default.stack-gcp: Object 'Kind' is missing in '{"level":"info","ts":1580980892.3018987,"logger":"stacks","msg":"Unpacking stack","to":"/dev/stdout"}'
```

I believe this is independent of hosted mode though.

Reply:
Ah good catch, thanks! I wonder if this was already an issue if run in debug mode - we used to log this at that level per - https://github.com/crossplaneio/crossplane/pull/1227/files#diff-8452f5d38b0987fceae61edde51d44ccL459. I'll revert this to debug mode for now.

Reply:
I've just confirmed I can install `stack-gcp` successfully now that the log line is at debug.